### PR TITLE
deploy: fixes for CI

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Ok to test

**Comment by matz-e** [comment]
Retest this please

**Comment by bbpadministrator** [comment]
retest this please

**Comment by bbpadministrator** [comment]
retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please (one more time)

**Comment by matz-e** [comment]
Retest this please

File: var/spack/repos/builtin/packages/mod2c/package.py [hunk 1]
```diff
@@ -26,5 +26,5 @@ def cmake_args(self):
             options.append('-DCMAKE_BUILD_WITH_INSTALL_RPATH=1')
         return options
 
-    def setup_environment(self, spack_env, run_env):
-        run_env.set('MODLUNIT', join_path(self.prefix, 'share/nrnunits.lib'))
+    def setup_run_environment(self, env):
+        env.set('MODLUNIT', join_path(self.prefix, 'share/nrnunits.lib'))
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/mod2c/package.py:30]: Oops..didn't know this was pending. Anyway, I will remove this package ins separate PR (it's now integrated into coreneuron).

File: var/spack/repos/builtin/packages/py-numpy/package.py [hunk 2]
```diff
@@ -113,12 +113,6 @@ class PyNumpy(PythonPackage):
     # GCC 4.8 is the minimum version that works
     conflicts('%gcc@:4.7', msg='GCC 4.8+ required')
 
-    def setup_dependent_package(self, module, dependent_spec):
-        if 'python' in self.spec:
-            python_version = self.spec['python'].version.up_to(2)
-        else:
-            python_version = dependent_spec['python'].version.up_to(2)
-
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-numpy/package.py:121]: Just curious without knowing the impact : Is this also in upstream? (thinking if this being used in any dependency packages)

File: var/spack/repos/builtin/packages/py-pip/package.py [hunk 3]
```diff
@@ -30,6 +30,5 @@ class PyPip(PythonPackage):
     # However, pip requires setuptools during runtime as well.
     depends_on('py-setuptools', type=('build', 'run'))
 
-    def setup_environment(self, spack_env, run_env):
-        run_env.prepend_path('PATH', self.spec.prefix.bin)
-
+    def setup_run_environment(self, env):
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-pip/package.py:33]: Curious : wasn't this already in the upstream ? i.e. how this was not done during initial merge with the upstream?

**Review by pramodk — COMMENTED** [review]
LGTM, some minor questions to clarify.

File: var/spack/repos/builtin/packages/mod2c/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/mod2c/package.py:30]: This was most likely a leftover, only checked when the file changed :)

File: var/spack/repos/builtin/packages/py-pip/package.py (hunk 3 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-pip/package.py:33]: Probably a merge artifact of ours.

File: var/spack/repos/builtin/packages/py-numpy/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-numpy/package.py:121]: Um, this code does not do anything at all. It's not in upstream, and all it does is to grab the python version. Probably a git merge thing I did not catch before.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-numpy/package.py:121]: Checked, this was only in our code, but as I mention above, is code that does not do anything.

File: var/spack/repos/builtin/packages/py-pip/package.py (hunk 3 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-pip/package.py:33]: Checked again, not in upstream. IIRC, you or @jorblancoa added that?

Comment (pramodk) [reply var/spack/repos/builtin/packages/py-pip/package.py:33]: hmmm....Ok. I don't remember.

**Review by pramodk — APPROVED** [review]

**Pull request merged**